### PR TITLE
Imgbb Actions

### LINKED ISSUE #1531: [ACTION] Imgbb new action
**Is there a specific app this action is for?**
Imgbb

**Please provide a link to the relevant API docs for the specific service / operation.**
https://api.imgbb.com/

**Upload a Photo** | Uploads a photo

### PR DESCRIPTION
**Pull request by vellames-turing**
Closes #1531

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Review by michelle0927 — DISMISSED** [review]
LGTM!

**Comment by michelle0927** [comment]
@vellames-turing I think you'll need to merge master with this branch to fix the Pull Request Checks.

**Comment by DilanAthukorala** [comment]
@vellames-turing Unable to upload images using base64 String. I am not getting any output the app just keeps loading.

image

**Comment by vellames-turing** [comment]
@DilanAthukorala Could you please send me your payload? I did some tests and I was able to send the image using the following base64

**iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAAApgAAAKYB3X3/OAAAABl0RVh0U29mdHdhcmUAd3d3Lmlua3NjYXBlLm9yZ5vuPBoAAANCSURBVEiJtZZPbBtFFMZ/M7ubXdtdb1xSFyeilBapySVU8h8OoFaooFSqiihIVIpQBKci6KEg9Q6H9kovIHoCIVQJJCKE1ENFjnAgcaSGC6rEnxBwA04Tx43t2FnvDAfjkNibxgHxnWb2e/u992bee7tCa00YFsffekFY+nUzFtjW0LrvjRXrCDIAaPLlW0nHL0SsZtVoaF98mLrx3pdhOqLtYPHChahZcYYO7KvPFxvRl5XPp1sN3adWiD1ZAqD6XYK1b/dvE5IWryTt2udLFedwc1+9kLp+vbbpoDh+6TklxBeAi9TL0taeWpdmZzQDry0AcO+jQ12RyohqqoYoo8RDwJrU+qXkjWtfi8Xxt58BdQuwQs9qC/afLwCw8tnQbqYAPsgxE1S6F3EAIXux2oQFKm0ihMsOF71dHYx+f3NND68ghCu1YIoePPQN1pGRABkJ6Bus96CutRZMydTl+TvuiRW1m3n0eDl0vRPcEysqdXn+jsQPsrHMquGeXEaY4Yk4wxWcY5V/9scqOMOVUFthatyTy8QyqwZ+kDURKoMWxNKr2EeqVKcTNOajqKoBgOE28U4tdQl5p5bwCw7BWquaZSzAPlwjlithJtp3pTImSqQRrb2Z8PHGigD4RZuNX6JYj6wj7O4TFLbCO/Mn/m8R+h6rYSUb3ekokRY6f/YukArN979jcW+V/S8g0eT/N3VN3kTqWbQ428m9/8k0P/1aIhF36PccEl6EhOcAUCrXKZXXWS3XKd2vc/TRBG9O5ELC17MmWubD2nKhUKZa26Ba2+D3P+4/MNCFwg59oWVeYhkzgN/JDR8deKBoD7Y+ljEjGZ0sosXVTvbc6RHirr2reNy1OXd6pJsQ+gqjk8VWFYmHrwBzW/n+uMPFiRwHB2I7ih8ciHFxIkd/3Omk5tCDV1t+2nNu5sxxpDFNx+huNhVT3/zMDz8usXC3ddaHBj1GHj/As08fwTS7Kt1HBTmyN29vdwAw+/wbwLVOJ3uAD1wi/dUH7Qei66PfyuRj4Ik9is+hglfbkbfR3cnZm7chlUWLdwmprtCohX4HUtlOcQjLYCu+fzGJH2QRKvP3UNz8bWk1qMxjGTOMThZ3kvgLI5AzFfo379UAAAAASUVORK5CYII=**

You need to remove the initials informations from the string, for example: **data:image/png;base64,**, if you have something similar in your string, please remove it :)

**Comment by DilanAthukorala** [comment]
@vellames-turing I think the issue is with large base64 Strings. When I use less detailed images with smaller base64 String it uploads correctly. But for large Strings it just keeps loading without giving any kind of output.

**Comment by vellames-turing** [comment]
@dannyroosevelt I am not able to test it. When I put a large base64, I have an infinity loading and no way to debug it.
I have noticed that I am receiving this exception on my local console
```
Uncaught (in promise) r {name: 'CombinedError', message: '[Network] ', graphQLErrors: Array(0), networkError: Error
    at https://pipedream.com/_static/npm.urql.b0ac1597.js:3:2452, response: Response}graphQLErrors: []message: "[Network] "name: "CombinedError"networkError: Error
    at https://pipedream.com/_static/npm.urql.b0ac1597.js:3:2452response: Response {type: 'cors', url: 'https://api.pipedream.com/graphql', redirected: false, status: 500, ok: false, …}[[Prototype]]: Error
start @ input.5b348b3b.js:1
await in start (async)
test @ input.5b348b3b.js:186
action @ builder-suspense.55169b40.js:1
v @ builder-suspense.55169b40.js:1
Be @ npm.vue.001be732.js:4
xe @ npm.vue.001be732.js:4
Wo @ npm.vue.001be732.js:4
onClick @ select-button.64ca93bd.js:1
(anonymous) @ select-button.64ca93bd.js:1
(anonymous) @ npm.vue.001be732.js:4
Be @ npm.vue.001be732.js:4
xe @ npm.vue.001be732.js:4
n @ npm.vue.001be732.js:4
i @ npm.logrocket.9d060a14.js:1
```

**Comment by dannyroosevelt** [comment]
> @dannyroosevelt I am not able to test it. When I put a large base64, I have an infinity loading and no way to debug it. I have noticed that I am receiving this exception on my local console
> 
> ```
> Uncaught (in promise) r {name: 'CombinedError', message: '[Network] ', graphQLErrors: Array(0), networkError: Error
>     at https://pipedream.com/_static/npm.urql.b0ac1597.js:3:2452, response: Response}graphQLErrors: []message: "[Network] "name: "CombinedError"networkError: Error
>     at https://pipedream.com/_static/npm.urql.b0ac1597.js:3:2452response: Response {type: 'cors', url: 'https://api.pipedream.com/graphql', redirected: false, status: 500, ok: false, …}[[Prototype]]: Error
> start @ input.5b348b3b.js:1
> await in start (async)
> test @ input.5b348b3b.js:186
> action @ builder-suspense.55169b40.js:1
> v @ builder-suspense.55169b40.js:1
> Be @ npm.vue.001be732.js:4
> xe @ npm.vue.001be732.js:4
> Wo @ npm.vue.001be732.js:4
> onClick @ select-button.64ca93bd.js:1
> (anonymous) @ select-button.64ca93bd.js:1
> (anonymous) @ npm.vue.001be732.js:4
> Be @ npm.vue.001be732.js:4
> xe @ npm.vue.001be732.js:4
> n @ npm.vue.001be732.js:4
> i @ npm.logrocket.9d060a14.js:1
> ```

I'm not sure what next steps are here. @vellames-turing @DilanAthukorala ?

**Comment by vellames-turing** [comment]
> > @dannyroosevelt I am not able to test it. When I put a large base64, I have an infinity loading and no way to debug it. I have noticed that I am receiving this exception on my local console
> > ```
> > Uncaught (in promise) r {name: 'CombinedError', message: '[Network] ', graphQLErrors: Array(0), networkError: Error
> >     at https://pipedream.com/_static/npm.urql.b0ac1597.js:3:2452, response: Response}graphQLErrors: []message: "[Network] "name: "CombinedError"networkError: Error
> >     at https://pipedream.com/_static/npm.urql.b0ac1597.js:3:2452response: Response {type: 'cors', url: 'https://api.pipedream.com/graphql', redirected: false, status: 500, ok: false, …}[[Prototype]]: Error
> > start @ input.5b348b3b.js:1
> > await in start (async)
> > test @ input.5b348b3b.js:186
> > action @ builder-suspense.55169b40.js:1
> > v @ builder-suspense.55169b40.js:1
> > Be @ npm.vue.001be732.js:4
> > xe @ npm.vue.001be732.js:4
> > Wo @ npm.vue.001be732.js:4
> > onClick @ select-button.64ca93bd.js:1
> > (anonymous) @ select-button.64ca93bd.js:1
> > (anonymous) @ npm.vue.001be732.js:4
> > Be @ npm.vue.001be732.js:4
> > xe @ npm.vue.001be732.js:4
> > n @ npm.vue.001be732.js:4
> > i @ npm.logrocket.9d060a14.js:1
> > ```
> 
> I'm not sure what next steps are here. @vellames-turing @DilanAthukorala ?

I can't debug it because I have an infinity load for large files. Is there some way to debug it?
Cc @js07

**Comment by vellames-turing** [comment]
I am moving this to `doing` again to implement these suggestions

File: components/imgbb/actions/upload-picture/upload-picture.mjs [hunk 1]
```diff
@@ -1,40 +1,48 @@
...
 import { stringify } from "qs";
+import { ConfigurationError } from "@pipedream/platform";
 
 export default {
-  key: "imgbb-upload-picture",
+  key: "imgbb-upload-picturesfojsfojsofjo",
   name: "Upload picture",
-  version: "0.2.1",
+  description: "Upload a picture to imgbb. [See the docs here](https://api.imgbb.com/)",
+  version: "0.2.3",
```
Comment (lcaresia) [inline components/imgbb/actions/upload-picture/upload-picture.mjs:10]:
```suggestion
  version: "0.2.2",
```

File: components/imgbb/imgbb.app.mjs [hunk 2]
```diff
@@ -1,11 +1,36 @@
...
+      };
+    },
+    _getAxiosParams(opts = {}) {
+      const res = {
+        ...opts,
+        url: this._getBaseUrl() + opts.path,
+        headers: this._getHeaders(),
+        params: {
+          ...opts.params,
+          key: this.$auth.api_key,
```
Comment (lcaresia) [inline components/imgbb/imgbb.app.mjs:23]: Move this to a getter method.

E.g.
```suggestion
          key: this._apiKey(),
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Just minor fixes.

**Review by lcaresia — APPROVED** [review]
LGTM!

**Pull request merged**